File: ui-preview/src/main/java/l/files/thumbnail/ApkThumbnailer.java
package l.files.thumbnail;

import android.content.Context;
import android.content.pm.PackageManager;

import java.io.IOException;

import l.files.fs.Path;
import l.files.ui.base.graphics.Rect;
import l.files.ui.base.graphics.ScaledBitmap;

import static l.files.ui.base.content.pm.Packages.getApkIconBitmap;

final class ApkThumbnailer implements Thumbnailer<Path> {

    @Override
    public boolean accepts(Path path, String mediaType) {
        return mediaType.equals("application/vnd.android.package-archive");
    }

    @Override
    public ScaledBitmap create(Path path, Rect max, Context context)
            throws IOException {
        PackageManager manager = context.getPackageManager();
        return getApkIconBitmap(path.toString(), max, manager);
    }
}

File: ui-preview/src/main/java/l/files/thumbnail/ApkThumbnailer.kt
package l.files.thumbnail

import android.content.Context
import l.files.ui.base.content.pm.Packages
import l.files.ui.base.graphics.Rect
import java.nio.file.Path

internal object ApkThumbnailer : Thumbnailer<Path> {

  override fun accepts(path: Path, type: String) =
    type == "application/vnd.android.package-archive"

  override fun create(input: Path, max: Rect, context: Context) =
    Packages.getApkIconBitmap(input.toString(), max, context.packageManager)
}
